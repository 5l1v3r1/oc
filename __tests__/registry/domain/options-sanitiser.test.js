const optionsSanitiser = require('../../../src/registry/domain/options-sanitiser');

jest.mock('../../../src/registry/domain/authentication', () => ({
  middleware: jest.fn(() => 'auth middleware has been assigned')
}));

describe('options sanitiser', () => {
  const scenarios = [
    {
      description: 'empty baseUrl to equal "/"',
      input: { baseUrl: '' },
      options: {
        key: 'baseUrl',
        value: '/'
      }
    },
    {
      description: 'verbosity 3 to equal 3',
      input: {
        baseUrl: '',
        verbosity: 3
      },
      options: {
        key: 'verbosity',
        value: 3
      }
    },
    {
      description:
        'customHeadersToSkipOnWeakVersion to equal customHeadersToSkipOnWeakVersion.toLowerCase()',
      input: {
        baseUrl: '',
        customHeadersToSkipOnWeakVersion: ['aaa', 'BBB', 'CCC-ccc']
      },
      options: {
        key: 'customHeadersToSkipOnWeakVersion',
        value: ['aaa', 'bbb', 'ccc-ccc']
      }
    },
    {
      description: 'fallbackRegistryUrl to equal fallbackRegistryUrl + "/"',
      input: {
        baseUrl: '',
        fallbackRegistryUrl: 'http://test-url.com'
      },
      options: {
        key: 'fallbackRegistryUrl',
        value: 'http://test-url.com/'
      }
    },
    {
      description: 'empty prefix to equal "/"',
      input: {
        baseUrl: ''
      },
      options: {
        key: 'prefix',
        value: '/'
      }
    },
    {
      description: 'prefix "/-/" to equal "/-/"',
      input: {
        baseUrl: '',
        prefix: '/-/'
      },
      options: {
        key: 'prefix',
        value: '/-/'
      }
    },
    {
      description: 'empty baseUrl and prefix "/-/" to equal baseUrl "/-/"',
      input: {
        baseUrl: '',
        prefix: '/-/'
      },
      options: {
        key: 'baseUrl',
        value: '/-/'
      }
    },
    {
      description: 'baseUrl "/" and prefix "/-/" to equal baseUrl "/-/"',
      input: {
        baseUrl: '/',
        prefix: '/-/'
      },
      options: {
        key: 'baseUrl',
        value: '/-/'
      }
    },
    {
      description:
        'publishAuth {} to equal beforePublish "auth middleware has been assigned"',
      input: {
        baseUrl: '',
        publishAuth: {}
      },
      options: {
        key: 'beforePublish',
        value: 'auth middleware has been assigned'
      }
    }
  ];

  scenarios.forEach(({ description, input, options }) => {
    test(description, () => {
      const output = optionsSanitiser(input);
      expect(output[options.key]).toEqual(options.value);
      expect(output).toMatchSnapshot();
    });
  });
});
