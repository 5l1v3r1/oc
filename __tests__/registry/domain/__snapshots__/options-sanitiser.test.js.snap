// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options sanitiser baseUrl "/" and prefix "/-/" to equal baseUrl "/-/" 1`] = `
Object {
  "baseUrl": "/-/",
  "beforePublish": [Function],
  "customHeadersToSkipOnWeakVersion": Array [],
  "discovery": true,
  "hotReloading": false,
  "port": undefined,
  "prefix": "/-/",
  "publishValidation": [Function],
  "tempDir": "./temp/",
  "timeout": 120000,
  "verbosity": 0,
}
`;

exports[`options sanitiser customHeadersToSkipOnWeakVersion to equal customHeadersToSkipOnWeakVersion.toLowerCase() 1`] = `
Object {
  "baseUrl": "/",
  "beforePublish": [Function],
  "customHeadersToSkipOnWeakVersion": Array [
    "aaa",
    "bbb",
    "ccc-ccc",
  ],
  "discovery": true,
  "hotReloading": false,
  "port": undefined,
  "prefix": "/",
  "publishValidation": [Function],
  "tempDir": "./temp/",
  "timeout": 120000,
  "verbosity": 0,
}
`;

exports[`options sanitiser empty baseUrl and prefix "/-/" to equal baseUrl "/-/" 1`] = `
Object {
  "baseUrl": "/-/",
  "beforePublish": [Function],
  "customHeadersToSkipOnWeakVersion": Array [],
  "discovery": true,
  "hotReloading": false,
  "port": undefined,
  "prefix": "/-/",
  "publishValidation": [Function],
  "tempDir": "./temp/",
  "timeout": 120000,
  "verbosity": 0,
}
`;

exports[`options sanitiser empty baseUrl to equal "/" 1`] = `
Object {
  "baseUrl": "/",
  "beforePublish": [Function],
  "customHeadersToSkipOnWeakVersion": Array [],
  "discovery": true,
  "hotReloading": false,
  "port": undefined,
  "prefix": "/",
  "publishValidation": [Function],
  "tempDir": "./temp/",
  "timeout": 120000,
  "verbosity": 0,
}
`;

exports[`options sanitiser empty prefix to equal "/" 1`] = `
Object {
  "baseUrl": "/",
  "beforePublish": [Function],
  "customHeadersToSkipOnWeakVersion": Array [],
  "discovery": true,
  "hotReloading": false,
  "port": undefined,
  "prefix": "/",
  "publishValidation": [Function],
  "tempDir": "./temp/",
  "timeout": 120000,
  "verbosity": 0,
}
`;

exports[`options sanitiser fallbackRegistryUrl to equal fallbackRegistryUrl + "/" 1`] = `
Object {
  "baseUrl": "/",
  "beforePublish": [Function],
  "customHeadersToSkipOnWeakVersion": Array [],
  "discovery": true,
  "fallbackRegistryUrl": "http://test-url.com/",
  "hotReloading": false,
  "port": undefined,
  "prefix": "/",
  "publishValidation": [Function],
  "tempDir": "./temp/",
  "timeout": 120000,
  "verbosity": 0,
}
`;

exports[`options sanitiser prefix "/-/" to equal "/-/" 1`] = `
Object {
  "baseUrl": "/-/",
  "beforePublish": [Function],
  "customHeadersToSkipOnWeakVersion": Array [],
  "discovery": true,
  "hotReloading": false,
  "port": undefined,
  "prefix": "/-/",
  "publishValidation": [Function],
  "tempDir": "./temp/",
  "timeout": 120000,
  "verbosity": 0,
}
`;

exports[`options sanitiser publishAuth {} to equal beforePublish "auth middleware has been assigned" 1`] = `
Object {
  "baseUrl": "/",
  "beforePublish": "auth middleware has been assigned",
  "customHeadersToSkipOnWeakVersion": Array [],
  "discovery": true,
  "hotReloading": false,
  "port": undefined,
  "prefix": "/",
  "publishAuth": Object {},
  "publishValidation": [Function],
  "tempDir": "./temp/",
  "timeout": 120000,
  "verbosity": 0,
}
`;

exports[`options sanitiser verbosity 3 to equal 3 1`] = `
Object {
  "baseUrl": "/",
  "beforePublish": [Function],
  "customHeadersToSkipOnWeakVersion": Array [],
  "discovery": true,
  "hotReloading": false,
  "port": undefined,
  "prefix": "/",
  "publishValidation": [Function],
  "tempDir": "./temp/",
  "timeout": 120000,
  "verbosity": 3,
}
`;
